### ISSUE TITLE
Does Catalyst support multi-agent environments?

### ISSUE THREAD

**Issue by rfali** [catalyst-team/catalyst#1200]
Hi

1. Can Catalyst implementations be used in multi-agent environments?
2. Does Catalyst support vectorized environments?

I am more specifically talking about DQN/Rainbow (and not interested in PPO etc)  as my action domain is discrete.

### FAQ
Please review the FAQ before submitting an issue:
- [x] I have read the [documentation and FAQ](https://catalyst-team.github.io/catalyst/)
- [ ] I have reviewed the [minimal examples section](https://github.com/catalyst-team/catalyst#minimal-examples)
- [ ] I have checked the [changelog](https://github.com/catalyst-team/catalyst/blob/master/CHANGELOG.md) for main framework updates
- [ ] I have read the [contribution guide](https://github.com/catalyst-team/catalyst/blob/master/CONTRIBUTING.md)
- [ ] I have joined [Catalyst slack (#__questions channel)](https://join.slack.com/t/catalyst-team-core/shared_invite/zt-d9miirnn-z86oKDzFMKlMG4fgFdZafw) for issue discussion

**Comment by Scitator** [comment]
Hi!

Catalyst.RL 2021 is currently in the development stage, you could check [the demo notebook here](https://github.com/catalyst-team/catalyst/blob/master/examples/notebooks/demo%2021xx.ipynb).
tl;dr
- you could implement any architecture you would like
- and any policy logic you want
- connect all these through the custom "GameCallback"
- and watch you training :)

By saying this, I mean, that it's on your side to write the environment logic and how to sample trajectories from it (vectorized environments, etc). Catalyst is a high-level experiment runner on top of PyTorch and correctly focused on Deep Learning (CV/NLP/RecSys) and Representation Learning mostly. Reinforcement Learning is under development stage - you could run Reinforcement Learning with Catalyst, but it still requires a lot of custom code from your side.

You could also check [Catalyst.RL 2020](https://github.com/catalyst-team/catalyst-rl) and [an introduction for your own environment usage](https://medium.com/pytorch/robotic-assembly-using-deep-reinforcement-learning-dfd9916c5ad7). Nevertheless, the 2020 version is frozen nowadays

**Comment by rfali** [comment]
Thanks @Scitator. I actually was looking towards Catalyst-RL because of this multi-headed DQN as per this [PR](https://github.com/catalyst-team/catalyst/pull/198) that I wanted to use in a multi-agent environment. 

I looked at this [catalyst atari example](https://github.com/catalyst-team/catalyst-rl/tree/master/examples/atari). I see that the multi-headed dqn is part of this [dqn.py](https://github.com/catalyst-team/catalyst-rl/blob/master/catalyst_rl/rl/offpolicy/algorithms/dqn.py), and the config file is [this](https://github.com/catalyst-team/catalyst-rl/blob/master/examples/atari/config_dqn.yml). In order to use multi-headed DQN, I only need to uncomment [line46](https://github.com/catalyst-team/catalyst-rl/blob/ec18ff4a58b6d00652f772231db8de86debb4b3d/examples/atari/config_dqn.yml#L46) right?

1. Can I use this DQN only on the Gym Atari, and not any other custom environments? 
2. If I can use it on other custom environments (similar to Atari, I am actually looking at a multi-agent Atari from [PettingZoo](https://www.pettingzoo.ml/atari)), can you show me how by an example code or reference to some other implemented work using catalyst-rl?


Thanks!

**Comment by Scitator** [comment]
Sorry, Catalyst.RL is deprecated for now – you could use it, run it and do all other interesting research stuff on your own.
Nevertheless, active support is not possible for free time. If you still would like to collaborate on Catalyst.RL, you could write us at team@catalyst-team.com.
Moreover, speaking of the open-source Catalyst.RL 2, you could check out new examples [here](https://github.com/catalyst-team/catalyst/tree/master/examples/reinforcement_learning).

**Comment by rfali** [comment]
I just wanted to use the catalyst v1 out of the box for one of its implementation and was wondering how can I use catalyst agents in other environments (gym-like but not Gym itself). the catalyst v2 does not have the specific implementation I am looking for.
Anyways, seems like I will have to figure this out myself. Thanks!
